lib/internal/repl.js

```diff
@@ -84,18 +96,26 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
-    if (data) {
+    if (data && data[0] === '['
+        // Check the last two characters in case there is a trailing newline
+        && /\]/.test(data.substr(-2))) {
       try {
         repl.history = JSON.parse(data);
         if (!Array.isArray(repl.history)) {
```

Comment:
I think we can drop this check.
With the new condition at lines 99-100, we are sure it is an array.